crates/starknet-server/tests/test_dump_and_load.rs

```diff
@@ -217,4 +247,171 @@ mod dump_and_load_tests {
+
+        #[cfg(windows)]
+        {
+            // To send SIGINT signal on windows, windows-kill is needed
+            let mut kill = Command::new("windows-kill")
+                .args(["-SIGINT", &devnet_dump.process.id().to_string()])
+                .spawn()
+                .unwrap();
+            kill.wait().unwrap();
+        }
+
+        #[cfg(unix)]
+        {
+            let mut kill = Command::new("kill")
+                .args(["-s", "SIGINT", &devnet_dump.process.id().to_string()])
+                .spawn()
+                .unwrap();
+            kill.wait().unwrap();
+        }
+
+        // file should not be created if there are no transactions
+        if Path::new(dump_file_name).exists() {
+            panic!(
+                "Could find the dump file but there were no transactions to dump {}",
+                dump_file_name
+            );
+        }
+    }
+
+    #[tokio::test]
+    async fn dump_endpoint_fail_with_wrong_request() {
+        let devnet_dump = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let dump_body = Body::from(
+            json!({
+                "test": ""
+            })
+            .to_string(),
+        );
+        let result = devnet_dump.post_json("/dump".into(), dump_body).await.unwrap();
+        assert_eq!(result.status(), 422);
+    }
+
+    #[tokio::test]
+    async fn load_endpoint_fail_with_wrong_request() {
+        let devnet_load = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let load_body = Body::from(
+            json!({
+                "test": ""
+            })
+            .to_string(),
+        );
+        let result = devnet_load.post_json("/load".into(), load_body).await.unwrap();
+        assert_eq!(result.status(), 422);
+    }
+
+    #[tokio::test]
+    async fn dump_endpoint_fail_with_wrong_file_name() {
+        let devnet_dump = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let dump_body = Body::from(
+            json!({
+                "path": "///"
+            })
+            .to_string(),
+        );
+        let result = devnet_dump.post_json("/dump".into(), dump_body).await.unwrap();
+        assert_eq!(result.status(), 500);
+    }
+
+    #[tokio::test]
+    async fn load_endpoint_fail_with_wrong_path() {
+        let load_file_name = "load_file_name";
+        let devnet_load = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        let load_body = Body::from(
+            json!({
+                "path": load_file_name
+            })
+            .to_string(),
+        );
+        let result = devnet_load.post_json("/load".into(), load_body).await.unwrap();
+        assert_eq!(result.status(), 400);
+    }
+
+    #[tokio::test]
+    async fn dump_load_endpoints_transaction_and_state_after_load_is_valid() {
+        let dump_file_name = "dump_endpoint";
+        let dump_file_name_custom_path = "dump_endpoint_custom_path";
```

Comment:
No need to define the variable so early.

Reply:
I wanted to have them near, so I could be sure that they were different. Also, this will change during https://github.com/0xSpaceShard/starknet-devnet-rs/issues/205

Reply:
Ok, makes sense. You can resolve this conversation if you want.